### PR TITLE
Adding isValid to ensure the curl connection pool will return an usable connection

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: #1716

### CODE REVIEW

File: sdk/core/azure-core/src/http/curl/curl.cpp [hunk 1]
```diff
@@ -1100,6 +1102,12 @@ inline std::string GetConnectionKey(std::string const& host, CurlTransportOption
 }
 } // namespace
 
+bool CurlConnection::IsValid() const
+{
+  char buffer[32];
```
Comment (RickWinter) [inline sdk/core/azure-core/src/http/curl/curl.cpp:1107]: Do we need to read 32 bytes to perform this check?  Could we only peek 1 byte and achieve the same check?

Comment (vhvb1989) [reply sdk/core/azure-core/src/http/curl/curl.cpp:1107]: 1 byte would be enough, yes.

**Comment by vhvb1989** [comment]
# Performance results:

Azure VM:  Standard F8s_v2 (8 vcpus, 16 GiB memory)
Location: East US 2 (Zone 1)
Release project generated with: cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON ..

## Test parallel 1
azure-storage-blobs-performance downloadBlob --containerName=sample-container --blobName=sample-blob --connectionString=$AZURE_STORAGE_CONNECTION_STRING -w 30 -d 60

Running test: DownloadBlob
Description: Download a blob.

=== Global Options ===
{"Duration":60,
"Host":"",
"Insecure":false,
"Iterations":1,
"JobStatistics":false,
"Latency":false,
"NoCleanup":false,
"Parallel":1,
"Port":null,
"Rate":null,
"Warmup":30}

=== Test Options ===
{"BlobName":"sample-blob",
"ContainerName":"sample-container",
"connectionString":"***"}

### Results

With this PR's changes:
=== Results ===
Completed 205 operations in a weighted-average of 60s (3.401466 ops/s, 0.293991 s/op)

Without this PR's change:
=== Results ===
Completed 210 operations in a weighted-average of 60s (3.485589 ops/s, 0.286896 s/op)

## Test parallel 4

azure-storage-blobs-performance downloadBlob --containerName=sample-container --blobName=sample-blob --connectionString=$AZURE_STORAGE_CONNECTION_STRING -w 30 -d 60 -p4

Running test: DownloadBlob
Description: Download a blob.

=== Global Options ===
{"Duration":60,
"Host":"",
"Insecure":false,
"Iterations":1,
"JobStatistics":false,
"Latency":false,
"NoCleanup":false,
"Parallel":4,
"Port":null,
"Rate":null,
"Warmup":30}

=== Test Options ===
{"BlobName":"sample-blob",
"ContainerName":"sample-container",
"connectionString":"***"}

### Results

With this PR's change
=== Results ===
Completed 841 operations in a weighted-average of 60s (13.965717 ops/s, 0.0716039 s/op)

Without this PR's change:
=== Results ===
Completed 843 operations in a weighted-average of 60s (14.019081 ops/s, 0.0713313 s/op)


## Test parallel 8

azure-storage-blobs-performance downloadBlob --containerName=sample-container --blobName=sample-blob --connectionString=$AZURE_STORAGE_CONNECTION_STRING -w 30 -d 60 -p 8

Running test: DownloadBlob
Description: Download a blob.


=== Global Options ===
{"Duration":60,
"Host":"",
"Insecure":false,
"Iterations":1,
"JobStatistics":false,
"Latency":false,
"NoCleanup":false,
"Parallel":8,
"Port":null,
"Rate":null,
"Warmup":30}

=== Test Options ===
{"BlobName":"sample-blob",
"ContainerName":"sample-container",
"connectionString":"***"}

### Results

With this PR's change:
=== Results ===
Completed 1,663 operations in a weighted-average of 60s (27.661909 ops/s, 0.0361508 s/op)

Without this PR's change:
=== Results ===
Completed 1,695 operations in a weighted-average of 60s (28.171612 ops/s, 0.0354967 s/op)

**Comment by vhvb1989** [comment]
closing on preference of: https://github.com/Azure/azure-sdk-for-cpp/pull/1721/files

**Pull request closed without merging**